File: Behavioral/Lab14/Typescript/src/Entities/realtor_room.ts
import { Customer } from "./customer";
import { RealEstate } from "./real_estate";
import { RealtorAbstract } from "./realtor";
import { Room } from "./room";

export class RealtorRoom extends RealtorAbstract {
    public override SellEstate(customer: Customer, estate: RealEstate): void {
        if (estate instanceof Room) {
            console.log(`${this.nameAndSurname} is selling ${estate.constructor.name} to ${customer.nameAndSurname} with price ${estate.priceInUSD}$`);
        } else if (this._realtor !== null) {
            this._realtor.SellEstate(customer, estate);
        }
    }
}

File: Behavioral/Lab14/C#/Entities/RealtorRoom.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace C_.Entities
{
    public class RealtorRoom : RealtorAbstract
    {
        public RealtorRoom(string nameAndSurname, string phone, string email, double yearsOfWork) : base(nameAndSurname, phone, email, yearsOfWork)
        {
        }

        public override void SellEstate(Customer customer, RealEstate estate)
        {
            if(estate is Room)
                Console.WriteLine($"{NameAndSurname} is selling {estate.GetType().Name} to {customer.NameAndSurname} with price {estate.PriceInUSD}$");
            else if(_realtor != null)
                _realtor.SellEstate(customer, estate);
        }
    }
}
